### PR TITLE
add TextEditor::onDidPasteText

### PR DESCRIPTION
**Pull request by kandros**
Add TextEditor::onDidPasteText is order to subscribe to the paste event.

### CODE REVIEW

File: src/text-editor.coffee [hunk 1]
```diff
@@ -2848,6 +2858,7 @@ class TextEditor extends Model
 
       didInsertEvent = {text, range}
       @emitter.emit 'did-insert-text', didInsertEvent
+      @emitter.emit 'did-paste-text', didInsertEvent
```
Comment (ericcornelissen) [inline src/text-editor.coffee]: What does this new feature exactly add, since all it seems to do is emit the same message as `'did-insert-text'`?

**Review by ghost — APPROVED** [review]

**Comment by aminya** [comment]
The code should be updated:
https://github.com/atom/atom/blob/a239092460e9339ee4d56e5d73258ff7d04fd0d8/spec/text-editor-spec.js#L5298:
```js
     it("notifies ::onDidPasteText observers", function() {
          const insertedStrings = [];
          editor.onDidPasteText(({text, range}) => insertedStrings.push(text));

          atom.clipboard.write("hello");
          editor.pasteText();

          expect(insertedStrings).toEqual(["hello"]);
   });
```

https://github.com/atom/atom/blob/a239092460e9339ee4d56e5d73258ff7d04fd0d8/src/text-editor.js#L913:
```js
    // Extended: Calls your `callback` after text has been pasted.
    //
    // * `callback` {Function}
    //  * `event` event {Object}
    //    * `text` {String} text to be inserted
    //
    // Returns a {Disposable} on which `.dispose()` can be called to unsubscribe.
    onDidPasteText(callback) {
      return this.emitter.on('did-paste-text', callback);
    }
```

https://github.com/atom/atom/blob/a239092460e9339ee4d56e5d73258ff7d04fd0d8/src/text-editor.js#L1589:
```js
      this.emitter.emit('did-paste-text', didInsertEvent);
```

**Comment by kandros** [comment]
> The code should be updated:
> 
> https://github.com/atom/atom/blob/a239092460e9339ee4d56e5d73258ff7d04fd0d8/spec/text-editor-spec.js#L5298
> 
> :
> ```js
>      it("notifies ::onDidPasteText observers", function() {
>           const insertedStrings = [];
>           editor.onDidPasteText(({text, range}) => insertedStrings.push(text));
> 
>           atom.clipboard.write("hello");
>           editor.pasteText();
> 
>           expect(insertedStrings).toEqual(["hello"]);
>    });
> ```
> 
> https://github.com/atom/atom/blob/a239092460e9339ee4d56e5d73258ff7d04fd0d8/src/text-editor.js#L913
> 
> :
> ```js
>     // Extended: Calls your `callback` after text has been pasted.
>     //
>     // * `callback` {Function}
>     //  * `event` event {Object}
>     //    * `text` {String} text to be inserted
>     //
>     // Returns a {Disposable} on which `.dispose()` can be called to unsubscribe.
>     onDidPasteText(callback) {
>       return this.emitter.on('did-paste-text', callback);
>     }
> ```
> 
> https://github.com/atom/atom/blob/a239092460e9339ee4d56e5d73258ff7d04fd0d8/src/text-editor.js#L1589
> 
> :
> ```js
>       this.emitter.emit('did-paste-text', didInsertEvent);
> ```

I no longer use the editor, I have enabled “allows edit from mantainers“ for this PR feel free to update it or close it

**Comment by sadick254** [comment]
It is not clear what functionality this adds. No need to emit another event in this case `did-insert-text` would suffice.

**Pull request closed without merging**